libraries/apollo-compiler/src/main/kotlin/com/apollographql/apollo3/compiler/ir/IrSchemaBuilder.kt

```diff
@@ -38,9 +38,6 @@ object IrSchemaBuilder {
       val typeDefinition = schema.typeDefinition(name)
 
       when {
-        typeDefinition.isBuiltIn() && typeDefinition !is GQLScalarTypeDefinition-> {
-          // do nothing, introspection types are in apollo-api
-        }
```

Comment:
If you're doing this, you'll have to remove `CompiledSchemaType` and friends [there](https://github.com/apollographql/apollo-kotlin/blob/a4d5f82888218fddf1d70ee9bdbbfe17befb04a2/libraries/apollo-api/src/commonMain/kotlin/com/apollographql/apollo3/api/CompiledGraphQL.kt#L378)

And also the shortcuts in codegen [there](https://github.com/apollographql/apollo-kotlin/blob/b0cac343c06c591273df779b4c423b3422a9af05/libraries/apollo-compiler/src/main/kotlin/com/apollographql/apollo3/compiler/codegen/kotlin/KotlinResolver.kt#L188). 

I'm all for it but if we want to be able to backport this to 3.x, maybe do this in a separate PR?

Reply:
I think I'm gonna revert this after all, it breaks `generateSchema` and not sure it's worth it

Reply:
This didn't work when using enums from the built-in introspection (so either `__TypeKind` or `__DirectiveLocation`. This fixes it, but not sure if it's odd.

Reply:
I guess that's OK for 3.x given we have a hack already for scalars. Certainly something to revisit for 4.x though

Reply:
Yeah I think this PR should be OK to merge in 3.x. I'll open other PRs for 4.x only (with some breaking changes).